app/app/src/main/kotlin/com/hedvig/app/feature/loggedin/ui/LoggedInActivity.kt

```diff
@@ -135,11 +134,9 @@ class LoggedInActivity : AppCompatActivity() {
       launch {
         lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
           authTokenService.authStatus.first { it is AuthStatus.LoggedIn }
-          reviewDialogViewModel.shouldOpenReviewDialog.collect { shouldOpenReviewDialog ->
-            if (shouldOpenReviewDialog) {
-              showReviewWithDelay()
-            }
-          }
+          waitUntilAppReviewDialogShouldBeOpenedUseCase.invoke()
+          delay(REVIEW_DIALOG_DELAY_MILLIS)
```

Comment:
why delay?

Reply:
We have had this in the past too.
The decision was made as a result of basically what I am saying here https://hedviginsurance.slack.com/archives/C03HT2JRDPG/p1709909630305479
A combination of not wanting to show the Dialog immediately as we are done with a flow, but before the member actually realizes that the flow is done (confusion)
But also it has to be a number which if possible will not overlap with if a member at the same time tries to click something else, so they accidentally dismiss the review dialog. It's quite easy to happen surprisingly, I've done it many times in the past.